test/switchtoactivity_test.js

```diff
@@ -0,0 +1,75 @@
+suite('switchToActivity', function() {
+  var launch = require('../lib/launch').launch;
+  var switchToApp = require('../lib/switchtoapp').switchToApp;
+  var switchToActivity = require('../lib/switchtoactivity').switchToActivity;
+
+  var ACTIVITY_CALLER_DOMAIN = 'activity.caller.gaiamobile.org';
+  var ACTIVITY_CALLEE_DOMAIN = 'activity.callee.gaiamobile.org';
+
+  var apps = {};
+  apps[ACTIVITY_CALLER_DOMAIN] = __dirname + '/apps/activitycaller';
+  apps[ACTIVITY_CALLEE_DOMAIN] = __dirname + '/apps/activitycallee';
+
+  var profile = {
+    settings: {
+      'ftu.manifestURL': null,
+      'lockscreen.enabled': false,
+      'screen.timeout': 0
+    },
+    // Need this so that swithToApp can also focus on the app frame.
+    prefs: {
+      'focusmanager.testmode': true
+    },
+    apps: apps
+  };
+
+  var client = createClient(profile);
+  marionette.plugin('mozApps', require('../lib/apps'));
+
+  suite('switch to app that is run as inline activity', function() {
+    var activityCallerOrigin = 'app://' + ACTIVITY_CALLER_DOMAIN;
+    var activityCalleeOrigin = 'app://' + ACTIVITY_CALLEE_DOMAIN;
+
+    setup(function(done) {
+      launch(client.mozApps, activityCallerOrigin, done);
+    });
+
+    setup(function(done) {
```

Comment:
I have a mixed feelings about several ```setup``` functions, IMO it looks a bit more readable then 2+ nested callbacks, but can use any approach :)